spark/src/main/scala/org/apache/spark/sql/delta/files/DeltaSourceSnapshot.scala

```diff
@@ -52,51 +54,47 @@ class DeltaSourceSnapshot(
     (part, data)
   }
 
-  protected def initialFiles: Dataset[IndexedFile] = {
+  private lazy val cachedState = {
     import spark.implicits.rddToDatasetHolder
     import org.apache.spark.sql.delta.implicits._
 
+    val numPartitions = snapshot.getNumPartitions
+
     cacheDS(
-      snapshot.allFiles.sort("modificationTime", "path")
+      snapshot.allFiles
+        // This allows us to control the number of partitions created from the sort instead of
+        // using the shufflePartitions setting
+        .repartitionByRange(numPartitions, col("modificationTime"), col("path"))
+        .sort("modificationTime", "path")
         .rdd.zipWithIndex()
         .toDF("add", "index")
+        // Stats aren't used for streaming reads right now, so decrease
+        // the size of the files by nulling out the stats if they exist
+        .withColumn("add", col("add").withField("stats", DataSkippingReader.nullStringLiteral))
```

Comment:
@zsxwing  This change is ok for DBR, right? At least, `DeltaLog.filterFileList` only seems to use partition filters.